src/content/operations/nat-traversal/_index.en.md

```diff
@@ -0,0 +1,415 @@
++++
+title = "NAT Traversal"
+date = 2020-08-12T16:02:00+02:00
+weight = 25
++++
+
+## Basic Overview
+
+Submariner establishes the dataplane tunnels between clusters over port
+`4500/udp` by default,  that port can be customized per cluster and per gateway,
+and it's published as part of the `Endpoint` objects in the broker, as `udp-port`
+inside of the `BackendConfig` field.
+
+### Public vs Private IP
+
+`Endpoint` objects publish the reachable IPs discovered by the gateway. A
+`PrivateIP` which is the IP assigned an interface on the gateway node where
+that `Endpoint` belongs, and `PublicIP` which is the src IP for the packets
+sent from the gateway to the Internet, currently we discover such IP via
+[ipify.org](https://ipify.org) but other mechanisms will be added in the future.
+
+### Reachability
+
+For two endpoints to connect to each other, at least one of them should be reachable
+either on the public or private IP address and firewall configuration should allow the
+tunnel encapsulation port.
+Using the `gateway.submariner.io/preferred-server=true` label only the labeled side
+needs to be reachable to other endpoints. For example:
+
+```bash
+kubectl label node $NODE gateway.submariner.io/preferred-server=true
+```
+
+Each gateway implements a UDP NATT discovery protocol where each gateway queries the
+gateways of the remote clusters on both the public & private IPs, figuring out the
+most suitable IP for the cable driver to use with preference on the private IP
+and the NAT characteristics of the path.
+
+This protocol is disabled by default, and should be enabled by assigning an UDP
+port to each gateway node on each cluster.
+
+```bash
+kubectl label node $NODE gateway.submariner.io/natt-discovery-port=4490
+```
+
+If the protocol has been enabled, and no reachability is possible between the
+endpoints the default settings will be used (which can be controlled with the
+NAT field of the operator or --natt parameter during subctl join), `--natt=false`
+will attempt the connection using the PrivateIP, `--natt=true` will attempt the
+connection using the PublicIP.
+
+### IP selection algorithm
+
+The following flow chart describes the IP selection algorithm:
+
+<!-- source: https://app.diagrams.net/#G1e_M84l8iQH6U4QqD3-Jw2ypIWiRfLQkk -->
+![Public/Private IP selection](/images/natt/public_private_selection_algorithm.svg)
```

Comment:
This diagram is cool but I wonder if we should include this much implementation detail in the end user guide. This can easily become stale - if the algorithm is changed in any way then we'd have to remember to update this diagram which may be problematic.

Reply:
It can be helpful for devs but we don't have implementation flow diagrams for other components on the website. I think these are more suited for the individual projects.

Reply:
> ...hopefully this algorithm shouldn't really change much...
> 

It would a bit if we end up keeping https://github.com/submariner-io/submariner/pull/1209 😄

Reply:
But the behavior is the same right?, we just change how it's really implemented, and when do we notify the cable engine of the public or privateIP. Still the overall behavior is the same.

I still it can be helpful to admins to understand what's going on and what do we make us decide between one IP or the other.

I can remove it too if we think it's better to keep it off.

@nyechiel @skitt what are your opinions ^ ?  I'm fine with either :)

Reply:
Since it's a fairly big change in a core area, I think it makes sense to include all that info for now. I hope that we can trim it down once it's becoming more mature and we learn from users what is trivial and what not. If we remove it at some point, let's ensure it's not going away and included for e.g within the enhancement repo.

Reply:
++ to keep it as @nyechiel mentioned.